app/src/main/java/com/example/mysimplecoindeck/ui/fragments/SingleCoinFragment.kt

```diff
@@ -19,38 +24,62 @@ class SingleCoinFragment: Fragment(R.layout.fragment_coin_detail) {
     private val viewModel: CoinsViewModel by viewModels()
     private lateinit var binding: FragmentCoinDetailBinding
     private val args: SingleCoinFragmentArgs by navArgs()
+    private lateinit var coin: Coin
+    private var uiDetailStateJob: Job? = null
```

Comment:
Okay for now, but I think it would be better to keep the Fragments/ Views free from state (except UI state like scroll position etc). This can be achieved by adding a `SingleCoinViewModel` that holds the coin data.